spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRowLevelOperationsWithLineage.java

```diff
@@ -399,8 +401,8 @@ public void testDelete() throws NoSuchTableException, ParseException, IOExceptio
   private List<Object[]> rowsWithLineageAndFilePos() {
     return sql(
         "SELECT s._last_updated_sequence_number, s.id, s.data, s._row_id, files.first_row_id, s._pos FROM %s"
-            + " AS s JOIN %s.files AS files ON files.file_path = s._file ORDER BY s._row_id",
-        selectTarget(), selectTarget());
+            + " AS s JOIN %s.files VERSION AS OF '%s' AS files ON files.file_path = s._file ORDER BY s._row_id",
+        selectTarget(), tableName, branch != null ? branch : "main");
```

Comment:
Needed because with the inherited avro test is the first time we're testing this with branching and so validation of newly inserted rows needs to time travel to get the files for a particular branch